packages/@uppy/companion/src/server/provider/dropbox/index.js

```diff
@@ -117,14 +139,26 @@ class DropBox extends Provider {
 
   async size ({ id, token }) {
     return this.#withErrorHandling('provider.dropbox.size.error', async () => {
-      const { size } = await (await getClient({ token })).post('files/get_metadata', { json: { path: id }, responseType: 'json' }).json()
+      // Fetch rootNamespaceId if not already set
+      if (!this.rootNamespaceId) {
+        const userInfoResponse = await userInfo({ token })
+        this.rootNamespaceId = userInfoResponse.root_info?.root_namespace_id
+      }
+
+      const { size } = await (await getClient({ token, rootNamespaceId: this.rootNamespaceId })).post('files/get_metadata', { json: { path: id }, responseType: 'json' }).json()
```

Comment:
to avoid all this duplication we should probably put `await userInfo({ token })` inside `getClient`